Model/Api/Client.php

```diff
@@ -26,53 +32,68 @@ class Client
     /** @var Configuration */
     private $config;
 
+    /**
+     * @var ResponseFactory
+     */
+    private $responseFactory;
+
+    /**
+     * @var ClientFactory
+     */
+    private $clientFactory;
+
     /**
      * Client constructor.
      * @param Json $json
      * @param Logger $logger
      * @param Configuration $config
+     * @param ClientFactory $clientFactory
+     * @param ResponseFactory $responseFactory
      */
     public function __construct(
         Json $json,
         Logger $logger,
-        Configuration $config
+        Configuration $config,
+        ClientFactory $clientFactory,
+        ResponseFactory $responseFactory
     ) {
         $this->json = $json;
         $this->logger = $logger;
         $this->config = $config;
+        $this->clientFactory = $clientFactory;
+        $this->responseFactory = $responseFactory;
     }
 
     /**
      * @param TokenInterface $token
      * @param Request $request
      * @return mixed
-     * @throws ClientResponseException
      * @throws ClientResponseErrorException
+     * @throws ClientResponseException
      */
     public function sendRequest(TokenInterface $token, Request $request)
     {
-        $json = $this->sendHttpRequest($token, $request);
-
-        $response = $this->json->unserialize($json);
-
-        if (!$response) {
-            throw new ClientResponseException(__('Error while receiving response from Allegro API'));
-        }
-
-        if (isset($response['errors'])) {
-            $errors = [];
-            foreach ($response['errors'] as $error) {
-                $errors[] = $error['message'] != '' ? $error['message'] : $error['userMessage'];
+        try {
+            $json = $this->sendHttpRequest($token, $request);
+            $response = $this->json->unserialize($json);
+
+            if (isset($response['errors'])) {
+                $errors = [];
+                foreach ($response['errors'] as $error) {
+                    $errors[] = $error['message'] != '' ? $error['message'] : $error['userMessage'];
+                }
+                throw new ClientResponseErrorException(
+                    __(
+                        'Errors returned in received from Allegro API response: %1',
+                        implode(' ', $errors)
+                    )
+                );
             }
-            throw new ClientResponseErrorException(
-                __(
-                    'Errors returned in received from Allegro API response: %1',
-                    implode(' ', $errors)
-                )
-            );
-        }
 
-        return $response;
+            return $response;
+        } catch (GuzzleException $e) {
```

Comment:
which line throws the GuzzleException ? the
```
$json = $this->sendHttpRequest($token, $request);
```
if yes, then we should wrap just it in the try-catch.